### PR TITLE
fix(sorting): RHICOMPL-1883 sort string-based columns in lowercase

### PR DESCRIPTION
**Pull request by skateman**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: app/models/rule.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ class Rule < ApplicationRecord
   )
 
   SORTABLE_BY = {
-    title: :title,
+    title: Arel.sql('LOWER(rules.title)'),
```
Comment (akofink) [inline app/models/rule.rb:17]: why does this need `rules.`?

File: app/models/host.rb [hunk 2]
```diff
@@ -7,7 +7,7 @@ class Host < ApplicationRecord
   OS_MAJOR_VERSION = Arel.sql("system_profile->'operating_system'->'major'")
 
   SORTABLE_BY = {
-    name: :display_name,
+    name: Arel.sql('LOWER(display_name)'),
```
Comment (vkrizan) [inline app/models/host.rb:10]:
```
Host.arel_table[:display_name].lower
```

File: app/models/rule.rb (hunk 1 above)
Comment (skateman) [reply app/models/rule.rb:17]: Without this I'm getting:
```
ERROR:  column reference "title" is ambiguous
```

File: app/models/host.rb (hunk 2 above)
Comment (skateman) [reply app/models/host.rb:10]: That is not compatible with our hash-sorting format that is being built in the controller :thinking:

**Comment by vkrizan** [comment]
@skateman what are we trying to solve? It seems that we might fix this on a database level, without the need to call `LOWER`. (Sorry Jira was down)

**Comment by skateman** [comment]
@vkrizan how?

**Comment by skateman** [comment]
Okay, so we discussed this with @vkrizan and we think that the current sorting is intended for humans and that's what we want...

**Pull request closed without merging**